### PR TITLE
add docs for TaskCountStatsMonitor

### PR DESCRIPTION
**Pull request by tomscut**
#### Add docs for org.apache.druid.server.metrics.TaskCountStatsMonitor. This is useful for users to monitor task status.

<hr>

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

<hr>

### CODE REVIEW

**Comment by tomscut** [comment]
@jihoonson Hi, could you please review this PR?

File: docs/configuration/index.md [hunk 1]
```diff
@@ -353,6 +353,7 @@ The following monitors are available:
 |`org.apache.druid.segment.realtime.RealtimeMetricsMonitor`|Reports statistics on Realtime processes.|
 |`org.apache.druid.server.metrics.EventReceiverFirehoseMonitor`|Reports how many events have been queued in the EventReceiverFirehose.|
 |`org.apache.druid.server.metrics.QueryCountStatsMonitor`|Reports how many queries have been successful/failed/interrupted.|
+|`org.apache.druid.server.metrics.TaskCountStatsMonitor`|Reports how many tasks are success/failed/running/pending/waiting.|
```
Comment (suneet-s) [inline docs/configuration/index.md]: As someone new to the monitors, I think explaining why you would need this is helpful. Are there specific use cases where I should be monitoring the task counts? Maybe I want to set up alerting on failed jobs?

Comment (tomscut) [reply docs/configuration/index.md]: @suneet-s  Thank you very much for your reply! 

Some implementation classes are registered by default and we don't need to register them manually, such as **CacheMonitor**, **JettyMonitor** and **ExecutorServiceMonitor**. But some are not registered by default and need to be configured ourselves, such as **QueryCountStatsMonitor** and **TaskCountStatsMonitor**. 
Docs: https://druid.apache.org/docs/latest/operations/metrics.html
PR: https://github.com/apache/druid/pull/6657/files

I found a description for **QueryCountStatsMonitor**, but not for **TaskCountStatsMonitor**. The description of **TaskCountStatsMonitor** might have been missed, so I added.

In our scenario, kafka has a large number of indexing tasks and a large amount of data that needs to be consumed with low latency. Therefore, task monitoring is very important.

We deployed the druid cluster with ambari, so monitoring and alerting were implemented based on ambari-metric-collector.

**Comment by tomscut** [comment]
@jihoonson Hi, could you please review this PR? Thx!

File: docs/configuration/index.md (hunk 1 above)
Comment (jihoonson) [inline docs/configuration/index.md:356]: Suggest `... tasks are success ...` -> `... tasks have been succeeded ...`

**Comment by suneet-s** [comment]
Merging this change - I'll make the suggested edits in a follow up patch.

**Comment by tomscut** [comment]
Thanks @suneet-s for the commit.

**Pull request merged**